{"primaryContentSections":[{"declarations":[{"platforms":["iOS"],"tokens":[{"kind":"attribute","text":"@frozen"},{"text":" ","kind":"text"},{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"text":"SighticError","kind":"identifier"}],"languages":["swift"]}],"kind":"declarations"}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/sighticanalytics\/sighticerror"]}],"schemaVersion":{"minor":3,"patch":0,"major":0},"metadata":{"roleHeading":"Enumeration","role":"symbol","navigatorTitle":[{"text":"SighticError","kind":"identifier"}],"symbolKind":"enum","fragments":[{"kind":"keyword","text":"enum"},{"text":" ","kind":"text"},{"text":"SighticError","kind":"identifier"}],"title":"SighticError","externalID":"s:16SighticAnalytics0A5ErrorO","modules":[{"name":"SighticAnalytics"}]},"sections":[],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError"},"kind":"symbol","hierarchy":{"paths":[["doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics"]]},"abstract":[{"type":"text","text":"Error that occurs during the analysis of a test recording by the Sightic Analytics backend."}],"topicSections":[{"identifiers":["doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/backendHTTPError(_:)","doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/internalError(_:)","doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/invalidAPIKey","doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/noConnection"],"title":"Enumeration Cases"},{"identifiers":["doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/Equatable-Implementations","doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/Error-Implementations"],"generated":true,"title":"Default Implementations"}],"relationshipsSections":[{"title":"Conforms To","identifiers":["doc:\/\/com.sighticanalytics.SighticAnalytics\/SQ","doc:\/\/com.sighticanalytics.SighticAnalytics\/s5ErrorP","doc:\/\/com.sighticanalytics.SighticAnalytics\/SH","doc:\/\/com.sighticanalytics.SighticAnalytics\/s8SendableP"],"type":"conformsTo","kind":"relationships"}],"seeAlsoSections":[{"identifiers":["doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticRecordingError","doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticUserError"],"title":"Errors","generated":true}],"references":{"doc://com.sighticanalytics.SighticAnalytics/s8SendableP":{"type":"unresolvable","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/s8SendableP","title":"Swift.Sendable"},"doc://com.sighticanalytics.SighticAnalytics/documentation/SighticAnalytics/SighticError/Equatable-Implementations":{"type":"topic","url":"\/documentation\/sighticanalytics\/sighticerror\/equatable-implementations","abstract":[],"identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/Equatable-Implementations","kind":"article","title":"Equatable Implementations","role":"collectionGroup"},"doc://com.sighticanalytics.SighticAnalytics/documentation/SighticAnalytics/SighticError/internalError(_:)":{"kind":"symbol","url":"\/documentation\/sighticanalytics\/sighticerror\/internalerror(_:)","role":"symbol","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/internalError(_:)","abstract":[{"text":"Internal error with an associated error code that can be provided to Sightic Analytics for troubleshooting.","type":"text"}],"title":"SighticError.internalError(_:)","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"text":"internalError","kind":"identifier"},{"text":"(","kind":"text"},{"preciseIdentifier":"s:Si","text":"Int","kind":"typeIdentifier"},{"kind":"text","text":")"}],"type":"topic"},"doc://com.sighticanalytics.SighticAnalytics/SQ":{"type":"unresolvable","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/SQ","title":"Swift.Equatable"},"doc://com.sighticanalytics.SighticAnalytics/documentation/SighticAnalytics/SighticRecordingError":{"fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"SighticRecordingError"}],"title":"SighticRecordingError","url":"\/documentation\/sighticanalytics\/sighticrecordingerror","kind":"symbol","abstract":[{"text":"Error that occurs while the app user performs the test.","type":"text"}],"type":"topic","navigatorTitle":[{"kind":"identifier","text":"SighticRecordingError"}],"role":"symbol","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticRecordingError"},"doc://com.sighticanalytics.SighticAnalytics/documentation/SighticAnalytics/SighticError/noConnection":{"title":"SighticError.noConnection","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/noConnection","url":"\/documentation\/sighticanalytics\/sighticerror\/noconnection","kind":"symbol","fragments":[{"kind":"keyword","text":"case"},{"text":" ","kind":"text"},{"kind":"identifier","text":"noConnection"}],"type":"topic","abstract":[{"type":"text","text":"Connection with Sightic backend could not be established."}],"role":"symbol"},"doc://com.sighticanalytics.SighticAnalytics/SH":{"type":"unresolvable","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/SH","title":"Swift.Hashable"},"doc://com.sighticanalytics.SighticAnalytics/documentation/SighticAnalytics":{"type":"topic","title":"SighticAnalytics","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics","kind":"symbol","url":"\/documentation\/sighticanalytics","role":"collection","abstract":[{"text":"Predict whether a person is impaired due to alcohol consumption.","type":"text"}]},"doc://com.sighticanalytics.SighticAnalytics/documentation/SighticAnalytics/SighticError/Error-Implementations":{"abstract":[],"title":"Error Implementations","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/Error-Implementations","role":"collectionGroup","type":"topic","kind":"article","url":"\/documentation\/sighticanalytics\/sighticerror\/error-implementations"},"doc://com.sighticanalytics.SighticAnalytics/documentation/SighticAnalytics/SighticUserError":{"navigatorTitle":[{"text":"SighticUserError","kind":"identifier"}],"identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticUserError","type":"topic","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"SighticUserError"}],"url":"\/documentation\/sighticanalytics\/sighticusererror","title":"SighticUserError","kind":"symbol","role":"symbol","abstract":[{"text":"Error caused by the user during the alignment or test phases.","type":"text"}]},"doc://com.sighticanalytics.SighticAnalytics/documentation/SighticAnalytics/SighticError/backendHTTPError(_:)":{"title":"SighticError.backendHTTPError(_:)","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/backendHTTPError(_:)","url":"\/documentation\/sighticanalytics\/sighticerror\/backendhttperror(_:)","kind":"symbol","fragments":[{"kind":"keyword","text":"case"},{"text":" ","kind":"text"},{"text":"backendHTTPError","kind":"identifier"},{"kind":"text","text":"("},{"text":"Int","kind":"typeIdentifier","preciseIdentifier":"s:Si"},{"kind":"text","text":")"}],"type":"topic","abstract":[{"type":"text","text":"Sightic backend returned an HTTP error."}],"role":"symbol"},"doc://com.sighticanalytics.SighticAnalytics/documentation/SighticAnalytics/SighticError":{"fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"SighticError"}],"title":"SighticError","url":"\/documentation\/sighticanalytics\/sighticerror","kind":"symbol","abstract":[{"text":"Error that occurs during the analysis of a test recording by the Sightic Analytics backend.","type":"text"}],"type":"topic","navigatorTitle":[{"kind":"identifier","text":"SighticError"}],"role":"symbol","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError"},"doc://com.sighticanalytics.SighticAnalytics/s5ErrorP":{"type":"unresolvable","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/s5ErrorP","title":"Swift.Error"},"doc://com.sighticanalytics.SighticAnalytics/documentation/SighticAnalytics/SighticError/invalidAPIKey":{"type":"topic","abstract":[{"type":"text","text":"Customer unique API key is invalid."}],"role":"symbol","url":"\/documentation\/sighticanalytics\/sighticerror\/invalidapikey","fragments":[{"kind":"keyword","text":"case"},{"text":" ","kind":"text"},{"kind":"identifier","text":"invalidAPIKey"}],"kind":"symbol","identifier":"doc:\/\/com.sighticanalytics.SighticAnalytics\/documentation\/SighticAnalytics\/SighticError\/invalidAPIKey","title":"SighticError.invalidAPIKey"}}}